Add a DictionaryObjectModel

erikos opened the pull request:
This can be used by the WordOfTheDay app.

https://phabricator.endlessm.com/T17276

cosimoc commented on ekncontent/ekncontent/eknc-dictionary-object-model.c:
```diff
@@ -0,0 +1,202 @@
+  switch (prop_id)
+    {
+    case PROP_WORD:
+      g_clear_pointer (&priv->word, g_free);
```
Since these properties are construct-only, you should not need to free the old value before setting the new one.

cosimoc commented on ekncontent/ekncontent/eknc-dictionary-object-model.c:
```diff
@@ -0,0 +1,202 @@
+typedef struct {
+  gchar *word;
+  gchar *definition;
+  gchar *partOfSpeech;
```
Don't use camel-case for variable names here

erikos commented:
Thanks for the review, addressed the comments.

ptomato commented on ekncontent/ekncontent/eknc-dictionary-object-model.c:
```diff
@@ -0,0 +1,199 @@
+  /**
+   * EkncDictionaryObjectModel:word:
+   *
+   * The actual word of the day
```
These documentation comments and property descriptions should make sense in the broader context of the eos-knowledge-content documentation, and not be so specific to word-of-the-day

erikos commented:
Added some tests, passed "make distcheck" as well.

smspillaz commented on ekncontent/tests/ekncontent/testDictionaryObjectModel.js:
```diff
@@ -0,0 +1,44 @@
+        });
+    });
+
+    describe ('JSON-LD marshaler', function () {
```
whitespace

smspillaz commented on ekncontent/tests/ekncontent/testDictionaryObjectModel.js:
```diff
@@ -0,0 +1,44 @@
+
+const InstanceOfMatcher = imports.tests.InstanceOfMatcher;
+
+describe ('Dictionary Object Model', function () {
```
whitespace between `describe` and `(`. There's also multiple instances of whitespace before `function` and `(`, but I'm not sure if that's intentional.

smspillaz commented on ekncontent/tests/ekncontent/testDictionaryObjectModel.js:
```diff
@@ -0,0 +1,44 @@
+
+        it('should marshal properties', function () {
+            expect(dictionaryObject.word).toBe('entreaty');
+            expect(dictionaryObject.definition).toBe('An earnest request or petition; a plea.');
```
Also add a check here for `part-of-speech`

erikos commented:
Thanks Sam for the review, addressed your comments. About the whitespace, originally I did follow the style in the other tests (read copy and paste) but removed them now. Added a test for part-of-speech too.

erikos commented:
@ptomato anything else we need to address here? I added the tests. If you have the chance would be great to get feedback from you on this one.

ptomato commented on ekncontent/tests/ekncontent/testDictionaryObjectModel.js:
```diff
@@ -0,0 +1,45 @@
+        });
+
+        it ('should inherit properties set by parent class (ContentObjectModel)', function() {
+            expect(dictionaryObject.title).toBeDefined();
```
This is a bit ambiguous, it would be better to test that the actual title text is actually marshalled through the JSON-LD marshaler.

ptomato commented on ekncontent/tests/ekncontent/testDictionaryObjectModel.js:
```diff
@@ -0,0 +1,45 @@
+            expect(dictionaryObject.part-of-speech).toBe('noun');
+        });
+
+        it ('should inherit properties set by parent class (ContentObjectModel)', function() {
```
nitpick: no space here

ptomato commented on ekncontent/ekncontent/eknc-dictionary-object-model.c:
```diff
@@ -0,0 +1,199 @@
+   * The actual word
+   */
+  eknc_dictionary_object_model_props[PROP_WORD] =
+    g_param_spec_string ("word", "The actual word.",
```
nitpicks: The "nick" (2nd parameter) should be the same as the property name, only in human-readable form: this is what you would see as the label in Glade next to the widget that sets the property, for example. So here it should be `"Word"`. The "blurb" is what you would see as the tooltip in Glade.

erikos commented:
Thanks Philip for taking another look, addressed the comments.

ptomato commented:
@erikos Build failed on Jenkins, would you mind investigating?

erikos commented:
Argh, I did not re-run distcheck, sorry :/ https://github.com/endlessm/eos-knowledge-lib/pull/1471
